pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/DefaultRebalancePreChecker.java

```diff
@@ -138,4 +157,80 @@ private boolean checkIsMinimizeDataMovement(String rebalanceJobId, String tableN
       return false;
     }
   }
+
+  private String checkDiskUtilization(String tableNameWithType, Map<String, Map<String, String>> currentAssignment,
```

Comment:
Just calling out that the pre-check results will be changing a bit as done in https://github.com/apache/pinot/pull/15233, you'll have to rebase and pick up that change once it is merged